DATAUP-612: Add exact_match_on field to dynamic dropdown options

MrCreosote opened the pull request:
Tests pass locally

briehl reviewed (dismissed):
Looks like a ton of changes, but really just autogenerated code. That works.

briehl commented:
I forget, how does NMS get a version bump?

MrCreosote commented:
In the server - I completely missed that. Fixed. Also added some more instructions for how to get the tests running (which currently requires python 2.7, ew)
